Examples/Algorithms/Geant4/src/ParticleTrackingAction.cpp

```diff
@@ -59,8 +59,20 @@ void ActsExamples::ParticleTrackingAction::PostUserTrackingAction(
     return;
   }
 
-  auto particle =
-      convert(*aTrack, eventData.trackIdMapping.at(aTrack->GetTrackID()));
+  const auto barcode = eventData.trackIdMapping.at(aTrack->GetTrackID());
+
+  auto hasHits = eventData.particleHitCount.find(barcode) !=
+                     eventData.particleHitCount.end() and
+                 eventData.particleHitCount.at(barcode) > 0;
+
+  if (not m_cfg.keepParticlesWithoutHits and not hasHits) {
+    auto n = eventData.particlesInitial.erase(
+        ActsExamples::SimParticle{barcode, Acts::PdgParticle::eInvalid});
+    assert(n == 1);
```

Comment:
should we use an error log here?

Reply:
the problem I see with this is that in release build without our assert workaround this error case will be silent while the `ACTS_ERROR` will at least show up in the log

Reply:
But thats the thing with all `assert` statements...

Reply:
that is true but I mostly see this in a place where a logger is not available or might be expensive to use

but up to you

Reply:
I actually would leave it with the `assert`.